IA-3955: Ignore paging param on datasource api querry used to filter groups

butofleury opened the pull request:
What problem is this PR solving? Explain here in one sentence.

Related JIRA tickets : [IA-3955](https://bluesquare.atlassian.net/browse/IA-3955), [IA-3894](https://bluesquare.atlassian.net/browse/IA-3894)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Doc

Tell us where the doc can be found (docs folder, wiki, in the code...).

## Changes

Get all data source linked to projects on the account!

## How to test

From org unit groups page, try to filter on datasource and check in the browser network tab. The data source api query return all data source without page size. 

Also try from the browser to access to dropdown endpoints for:
- [Data Source](http://localhost:8081/api/datasources/dropdown/)
- [Source Version](http://localhost:8081/api/sourceversions/dropdown/)

You can also sort group by source version.

## Print screen / video

[DataSource_Source-Version-List-–-Django-REST-framework.webm](https://github.com/user-attachments/assets/6ad43559-774c-4bc1-9e80-5b5d7df40d1b)

[Sort-Groups-by-sourceVersion.webm](https://github.com/user-attachments/assets/ef3c9077-40af-4d09-91d4-4816d81212d3)



## Notes

Things that the reviewers should know:

- known bugs that are out of the scope of the PR
- other trade-offs that were made
- does the PR depends on a PR in [bluesquare-components](https://github.com/BLSQ/bluesquare-components)?
- should the PR be merged into another PR?

## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: empty instance pop up

Refs: IA-3665
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.


[IA-3955]: https://bluesquare.atlassian.net/browse/IA-3955?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[IA-3894]: https://bluesquare.atlassian.net/browse/IA-3894?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

butofleury committed:
Getting all data source used to filter on org unit groups

beygorghor reviewed (approved):
This is working but:
- We should totally remove pagination params rom the payload, we still have page=1
- like describe on the ticket we should use a dedicated url for the dropdown values, this api is returning way to much infos

butofleury committed:
Merge branch 'main' into IA-3955-Ignore-paging-param-on-datasource-api-querry-used-to-filter-groups

butofleury committed:
Adding dropdown endpoints for datasources and source versions

butofleury committed:
Use dropdown endpoints for datasources and source versions filters on group page

butofleury committed:
Testing datasource dropdown endpoint

butofleury committed:
Testing source version dropdown endpoint

butofleury commented:
> This is working but:
> 
>     * We should totally remove pagination params rom the payload, we still have page=1
> 
>     * like describe on the ticket we should use a dedicated url for the dropdown values, this api is returning way to much infos

Updated the PR by adding and using dropdown endpoint for:
- [Data sources](https://github.com/BLSQ/iaso/pull/1975/files#diff-4bce274c8566e1460c212254c00f01b5331f2aa24c587a9cf613fe7d86cc984f) 
- [Source version](https://github.com/BLSQ/iaso/pull/1975/files#diff-44a717819788fcb2d8e0064d89100402acfa09016934c85826b052dc19c071e8R179)

 
The [previous changes](https://github.com/BLSQ/iaso/pull/1975/commits/29f06540c378f5eb0a4635b27d3f59737d9c9376)  was breaking the data source page. But it's now fixed by the latest commits.

butofleury committed:
Allow to sort groups by source version

beygorghor commented on hat/assets/js/apps/Iaso/domains/dataSources/requests.js:
```diff
@@ -103,6 +103,39 @@ export const useDataSourceVersions = () => {
+                }),
+        },
+    );
+};
```
this is probably due to a bad typing somewhere i think, normally you don't have to cast it into a string

beygorghor reviewed (approved):
Thi s is ok for me, much better with a specific endpoint ! maybe a back-end review can be nice here @tdethier  ?

butofleury committed:
Merge branch 'main' into IA-3955-Ignore-paging-param-on-datasource-api-querry-used-to-filter-groups

tdethier commented on iaso/tests/api/test_data_sources.py:
```diff
@@ -185,3 +185,8 @@ def test_datasource_filters(self):
+    def test_dropdown_datasource(self):
+        self.client.force_authenticate(self.joe)
+        response = self.client.get("/api/datasources/dropdown/?order=name")
+        self.assertJSONResponse(response, 200)
```
I would add more asserts here: 
- at least check if the received number of data sources matches what was expected
- maybe check the details of some of them (just the first one for instance), provided that there is some stable order

tdethier commented on iaso/tests/api/test_source_versions.py:
```diff
@@ -93,3 +93,9 @@ def test_list(self):
+        self.user.user_permissions.set([Permission.objects.get(codename=permission._SOURCES)])
+        self.client.force_authenticate(self.user)
+        response = self.client.get("/api/sourceversions/dropdown/")
+        self.assertJSONResponse(response, 200)
```
Same comment about more asserts

tdethier commented on iaso/tests/api/test_source_versions.py:
```diff
@@ -93,3 +93,9 @@ def test_list(self):
+        self.user.user_permissions.set([Permission.objects.get(codename=permission._SOURCES)])
+        self.client.force_authenticate(self.user)
+        response = self.client.get("/api/sourceversions/dropdown/")
+        self.assertJSONResponse(response, 200)
```
I would also add more tests for both endpoints:
- calling without authentication
- calling without permissions
- maybe a test with a user from another account, to make sure that the initial data sources and source versions are not fetched

butofleury replied:
@tdethier , I've added more tests as you suggested: https://github.com/BLSQ/iaso/pull/1975/commits/5743c4a9e3f51e3f970dfa5087e962cb9d8a81ce

tdethier reviewed (approved):
I only checked the backend, but it looks like it's working fine :+1: 

However, I would definitely add more tests if you have time.

butofleury committed:
Adding more asserts to test details for drop down datasource endpoint

butofleury committed:
Adding more asserts to test source details and source without user authentication/permission for drop down datasource endpoint

butofleury committed:
Testing drop down datasource endpoint with user from another account

butofleury committed:
Getting datasource and source versions data from dropdown endpoints without transformation

butofleury committed:
Fix query keys for datasource and sourceversion drop down endpoint calls

tdethier commented on iaso/tests/api/test_source_versions.py:
```diff
@@ -93,3 +103,35 @@ def test_list(self):
+        self.assertEqual(version["id"], self.version2.pk)
+        self.assertEqual(version["data_source"], self.data_source2.pk)
+        self.assertEqual(version["number"], self.version2.number)
+        self.assertEqual(version["data_source_name"], self.data_source2.name)
```
I would also probably add a `self.assertEqual(len(data), 1)` like you have on line 112

butofleury replied:
Thanks, just added that assert case

tdethier commented on iaso/tests/api/test_data_sources.py:
```diff
@@ -185,3 +185,12 @@ def test_datasource_filters(self):
         self.assertEqual(len(data["sources"]), 1)
         self.assertEqual(data["sources"][0]["id"], self.data_source.pk)
+
+    def test_dropdown_datasource(self):
```
And I would probably add the additional tests on data sources too

butofleury replied:
Thanks, just added 2 more test for user without authentication and user with authentication but without permissions.

tdethier reviewed (approved):
Still nitpicking, if you have time for that, that's great. Otherwise that's probably fine :+1:

butofleury committed:
Testing result length dropdown source version endpoint

butofleury committed:
Testing dropdown data source end point with user without permission/authentication

tdethier reviewed (approved):
Awesome, good job :+1:
